examples/usecases/ranking_with_multitask_learning.ipynb

```diff
@@ -0,0 +1,1663 @@
+{
```

Comment:
<p>If I understand it correctly, this use case it to weight the loss of each sample given another column. </p><p> </p><p>I think the example explains the functionality, but using "age" or "watching_times" as an example is pretty confusing. I do not think we would use that in the real-world. I would rather change the example to provide a sample_weight column and provide a mapping of age to weights.</p><p> </p><p>E.g. </p><p>Age 0-17: 1.0, </p><p>Age: 18-25: 10.0, </p><p>Age: 25-35: 5.0, </p><p>Age: 35+ : 1.0</p><p> </p><p>With an example that we want to focus on our target group and weight their examples higher. I think that is easier to understand</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/948/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

Reply:
<p>Your are right. That example is confusing.  </p><p>I am using the schema for TenRec, I think creating another column and encoding it just for this sampling weight use case could be a detour.</p><p>I simplified this use case to a single task (below) and as watching_times is the count of times the user watches the video, it indicates the strength of the relevance of that item for the user, and it could be reasonable using that column as sample weight.</p><p> </p><pre class="ql-syntax" spellcheck="false">output_block = mm.BinaryOutput("click",
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; post=mm.ColumnBasedSampleWeight("watching_times")
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;)
</pre>

Reply:
I like that example, that make more sense